streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -188,7 +188,8 @@ private SourceNodeFactory(String name, String[] topics, Pattern pattern, Deseria
         @SuppressWarnings("unchecked")
         @Override
         public ProcessorNode build() {
-            return new SourceNode(name, nodeToSourceTopics.get(name), keyDeserializer, valDeserializer);
+            String[] sourceTopics = nodeToSourceTopics.get(name);
```

Comment:
I guess doing this incrementally will not work. What about one (or multiple) PR that add `final` wherever possible over all files (maybe package by package to split it up in multiple PRs)? It's dumb work, but I would volunteer...